proxy/envoy/config.go

```diff
@@ -303,9 +303,11 @@ func buildInboundFilters(instances []*model.ServiceInstance) (HTTPRouteConfigs,
 			http.VirtualHosts = append(http.VirtualHosts, host)
 		case model.ProtocolTCP:
 			cluster := buildInboundCluster(service.Hostname, endpoint.Port, port.Protocol)
-			route := buildTCPRoute(cluster, endpoint.Address, endpoint.Port)
 			config := tcpConfigs.EnsurePort(port.Port)
-			config.Routes = append(config.Routes, route)
+			config.Routes = append(config.Routes,
+				buildTCPRoute(cluster, []string{endpoint.Address}, endpoint.Port),
+				buildTCPRoute(cluster, []string{service.Address, endpoint.Address}, port.Port),
```

Comment:
Yes, that makes sense and matches my own expectation. But generating service:port endpoint:port routes from the existing model did not give the correct endpoint/port tuple for the endpoint address (see corrected example configuration below). Perhaps there is an issue with how I setup the port/targetPort in the k8s templates that differs from the http ports?

```
# Port 90 listener
{
  "cluster": "inbound:9090",
  "destination_ip_list": [
    "10.16.0.230/32"
  ],
  "destination_ports": "9090"  # Never matches from external-to-internal TCP traffic (e.g. B => A:90)
},
{
  "cluster": "inbound:9090",
  "destination_ip_list": [
    "10.19.242.187/32"
  ],
  "destination_ports": "90"
},
```